src/java.base/share/classes/sun/security/ssl/SSLExtensions.java

```diff
@@ -54,7 +54,19 @@
             ByteBuffer m, SSLExtension[] extensions) throws IOException {
         this.handshakeMessage = hm;
 
+        if (m.remaining() < 2) {
+            throw hm.handshakeContext.conContext.fatal(
+                    Alert.ILLEGAL_PARAMETER,
+                    "Incorrect extensions: no length field");
+        }
+
         int len = Record.getInt16(m);
+        if (len > m.remaining()) {
+            throw hm.handshakeContext.conContext.fatal(
+                    Alert.ILLEGAL_PARAMETER,
+                    "Insufficient extensions data");
+        }
```

Comment:
For both of these blocks the checks themselves look OK, but illegal_parameter I thought was more for cases where a field value is out of range or inconsistent with already negotiated parameters.  I would think that decode_error would be more appropriate to cases like this where the encoding is structurally incorrect and the length doesn't match the actual data size.

Reply:
Good catch!  Updated to use decode_error.